express-api/src/services/notifications/notificationServices.ts

```diff
@@ -355,10 +374,106 @@ const sendNotification = async (
+    case 'accepted':
+      return NotificationStatus.Accepted;
+    case 'pending':
+      return NotificationStatus.Pending;
+    case 'cancelled':
+      return NotificationStatus.Cancelled;
+    case 'failed':
+      return NotificationStatus.Failed;
+    case 'completed':
+      return NotificationStatus.Completed;
+    default:
+      return NotificationStatus.Failed;
+  }
+};
+
+const updateNotificationStatus = async (
+  notificationId: number,
+  user: User,
+  queryRunner?: QueryRunner,
+) => {
+  const query = queryRunner ?? AppDataSource.createQueryRunner();
+  try {
+    //await query.startTransaction(); // Start transaction if queryRunner is provided
+
+    const notification = await query.manager.findOne(NotificationQueue, {
+      where: { Id: notificationId },
+    });
+
+    if (!notification || Object.keys(notification).length === 0) {
+      throw new Error(`Notification with id ${notificationId} not found.`);
+    }
+
+    const statusResponse = await chesServices.getStatusByIdAsync(notification.ChesMessageId);
+
+    if (statusResponse) {
+      notification.Status = convertChesStatusToNotificationStatus(statusResponse.status);
+      notification.UpdatedOn = new Date();
+      notification.UpdatedById = user.Id;
+      const updatedNotification = await query.manager.save(NotificationQueue, notification);
+
+      return updatedNotification;
+    } else {
+      throw new Error(`Failed to retrieve status for notification with id ${notificationId}.`);
+    }
+  } catch (error) {
+    await query.rollbackTransaction(); // Rollback transaction on error
+
+    throw error; // Rethrow the error for handling elsewhere if needed
+  } finally {
+    if (!queryRunner) {
+      await query.release(); // Release queryRunner if it was created locally
+    }
+  }
+};
+
+const getProjectNotificationsInQueue = async (
+  filter: ProjectNotificationFilter,
+  user: User,
+): Promise<PageModel<NotificationQueue>> => {
+  const { projectId, pageNumber, pageSize } = filter;
+  const notifications = await AppDataSource.getRepository(NotificationQueue).find({
+    where: {
+      ProjectId: projectId,
+    },
+    skip: (pageNumber ?? 0) * (pageSize ?? 0),
+    take: pageSize ?? 0,
+    order: { SendOn: 'ASC' },
+  });
+
+  const updatedNotifications: NotificationQueue[] = [];
+
+  const pageModel: PageModel<NotificationQueue> = {
+    items: notifications,
+    pageNumber: pageNumber ?? 0,
+    pageSize: pageSize ?? 0,
+  };
+  for (const notification of notifications) {
+    // run the updates for notifications that are in Pending or Accepted status as the other statuses are final
+    if (
+      notification.Status === NotificationStatus.Pending ||
+      notification.Status === NotificationStatus.Accepted
+    ) {
+      const updatedNotification = await updateNotificationStatus(notification.Id, user);
+      updatedNotifications.push(updatedNotification);
+    } else {
+      updatedNotifications.push(notification);
+    }
+  }
```

Comment:
There's a few things I'm noticing here.
* updatedNotifications array's final value seems to go unused, I think that you need to include the results of updating these notifications in the return value of the function.
* updateNotificationStatus should not use await as there are no dependencies on the result of this call until the function returns, you should instead collect each promise in to an array and then await them with either Promise.all() or Promise.allSettled(), you can expect a pretty significant performance increase from this